packages/core/src/layout/Sidebar/UserSettings.tsx

```diff
@@ -53,41 +40,8 @@ export function SidebarUserSettings() {
     <>
       <SidebarUserProfile open={open} setOpen={setOpen} />
       <Collapse in={open} timeout="auto">
-        {providers.includes('google') && (
-          <OIDCProviderSettings
-            title="Google"
-            apiRef={googleAuthApiRef}
-            icon={Star}
-          />
-        )}
-        {providers.includes('github') && (
-          <OAuthProviderSettings
-            title="Github"
-            apiRef={githubAuthApiRef}
-            icon={Star}
-          />
-        )}
-        {providers.includes('gitlab') && (
-          <OAuthProviderSettings
-            title="Gitlab"
-            apiRef={gitlabAuthApiRef}
-            icon={Star}
-          />
-        )}
-        {providers.includes('okta') && (
-          <OIDCProviderSettings
-            title="Okta"
-            apiRef={oktaAuthApiRef}
-            icon={Star}
-          />
-        )}
-        {providers.includes('oauth2') && (
-          <OIDCProviderSettings
-            title="YourOrg"
-            apiRef={oauth2ApiRef}
-            icon={Star}
-          />
-        )}
+        {providerSettings}
```

Comment:
As an alternativ implementation we could check here for `null` and use `<DefaultProviderSettings />` instead. Than this change doesn't require any modification to the template or app.